kcbq-connector/src/test/java/com/wepay/kafka/connect/bigquery/convert/BigQuerySchemaConverterTest.java

```diff
@@ -628,6 +632,43 @@ public void testAllFieldsNullable() {
     com.google.cloud.bigquery.Schema bigQueryTestSchema =
         new BigQuerySchemaConverter(true).convertSchema(kafkaConnectTestSchema);
     assertEquals(bigQueryExpectedSchema, bigQueryTestSchema);
+  }
+
+  @Test
+  public void testSimpleRecursiveSchemaThrows() {
+    final String fieldName = "RecursiveField";
+
+    org.apache.avro.Schema recursiveAvroSchema = org.apache.avro.SchemaBuilder
+        .record("RecursiveItem")
+        .namespace("com.example")
+        .fields()
+        .name(fieldName)
+        .type().unionOf().nullType().and().type("RecursiveItem").endUnion()
+        .nullDefault()
+        .endRecord();
+    Schema connectSchema = new AvroData(100).toConnectSchema(recursiveAvroSchema);
```

Comment:
Can we put a brief comment here stating that the `AvroData` class is used because it can produce recursive schemas, but the Connect `SchemaBuilder` class cannot? Otherwise it might be a little confusing that we're going about this in a sort of roundabout fashion of first constructing an Avro schema, then converting it to a Connect schema, then finally passing it to the `BigQuerySchemaConverter`.